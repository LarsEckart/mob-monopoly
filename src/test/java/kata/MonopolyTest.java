package kata;

import com.github.larseckart.tcr.TestCommitRevertMainExtension;
import org.approvaltests.Approvals;
import org.approvaltests.StoryBoard;
import org.approvaltests.reporters.DelayedClipboardReporter;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(TestCommitRevertMainExtension.class)
@UseReporter({DiffReporter.class, LinuxKdiff3Reporter.class, DelayedClipboardReporter.class})
class MonopolyTest {

    @Test
    void testBasicPlay() {
        StoryBoard story = new StoryBoard();
        Monopoly monopoly = new Monopoly(3);
        story.addFrame(monopoly);
        story.addFrame(monopoly.move(5).buyProperty().endTurn());
        story.addFrame(monopoly.move(3).buyProperty().endTurn());
        story.addFrame(monopoly.move(3).endTurn());
        Approvals.verify(story);
    }

    @Test
    void testIncomeTaxAndChanceAndJail() {
        StoryBoard story = new StoryBoard();
        ChancePile.stack(new ChanceCard.BankCard());
        Monopoly monopoly = new Monopoly(3);
        story.addFrame(monopoly);
        story.addFrame(monopoly.move(10).endTurn());
        story.addFrame(monopoly.move(4).endTurn());
        story.addFrame(monopoly.move(7).endTurn());
        Approvals.verify(story);
    }

    @Test
    void testOtherChances() {
        StoryBoard story = new StoryBoard();
        ChancePile.stack(new ChanceCard.BankCard()
                , new ChanceCard.GoToStCharlesPlace()
                , new ChanceCard.GoBack3Spaces()
                , new ChanceCard.SpeedingFine()
                , new ChanceCard.ChairmanOfTheBoard()
                , new ChanceCard.GetOutOfJailFree()
                , new ChanceCard.AdvanceToNearestRailroad()
        );
        int chanceCards = ChancePile.size();
        Monopoly monopoly = new Monopoly(chanceCards);
        story.addFrame(monopoly);
        for (int i = 0; i < chanceCards; i++) {
            if (i == 1) {
                story.addFrame(monopoly.move(7).buyProperty().endTurn());
            } else {
                story.addFrame(monopoly.move(7).endTurn());
            }
        }

        Approvals.verify(story);
    }

    @Test
    void testPlayerDetails() {
        Player player = new Player("Player 1");

        assertThat(player.details()).isEqualTo("Player 1 ($1500)  ");
    }

    @Test
    void testPlayerLocation() {
        Player player = new Player("Player 1");

        assertThat(player.location()).isEqualTo(0);
    }

    @Test
    void testRollingDoubles() {
        StoryBoard story = new StoryBoard();
        Monopoly monopoly = new Monopoly(2);
        story.addFrame(monopoly);
        story.addFrame(monopoly
                .move(12, true).buyProperty()
                .move(12, true).buyProperty()
                .move(11, false).buyProperty()
                .endTurn());
        Approvals.verify(story);
    }


    // write a test to pass go
    // write a test for rolling three doubles in a row
    // write a test for trying to go without rolling doubles
}
