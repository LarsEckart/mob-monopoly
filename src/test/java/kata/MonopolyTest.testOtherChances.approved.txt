Initial:
|          GO           | [Player 1, Player 2, Player 3, Player 4, Player 5, Player 6, Player 7] |
| Mediterranean Avenue  |                                   []                                   |
|    Community Chest    |                                   []                                   |
|     Baltic Avenue     |                                   []                                   |
|      Income Tax       |                                   []                                   |
|   Reading Railroad    |                                   []                                   |
|    Oriental Avenue    |                                   []                                   |
|        Chance         |                                   []                                   |
|    Vermont Avenue     |                                   []                                   |
|  Connecticut Avenue   |                                   []                                   |
|         Jail          |                                   []                                   |
|   St. Charles Place   |                                   []                                   |
|   Electric Company    |                                   []                                   |
|     States Avenue     |                                   []                                   |
|    Virginia Avenue    |                                   []                                   |
| Pennsylvania Railroad |                                   []                                   |
|    St. James Place    |                                   []                                   |
|    Community Chest    |                                   []                                   |
|   Tennessee Avenue    |                                   []                                   |
|    New York Avenue    |                                   []                                   |
|     Free Parking      |                                   []                                   |
|    Kentucky Avenue    |                                   []                                   |
|        Chance         |                                   []                                   |
|    Indiana Avenue     |                                   []                                   |
|    Illinois Avenue    |                                   []                                   |
|   B. & O. Railroad    |                                   []                                   |
|    Atlantic Avenue    |                                   []                                   |
|    Ventnor Avenue     |                                   []                                   |
|      Water Works      |                                   []                                   |
|    Marvin Gardens     |                                   []                                   |
|      Go To Jail       |                                   []                                   |
|    Pacific Avenue     |                                   []                                   |
| North Carolina Avenue |                                   []                                   |
|    Community Chest    |                                   []                                   |
|  Pennsylvania Avenue  |                                   []                                   |
|  Short Line Railroad  |                                   []                                   |
|        Chance         |                                   []                                   |
|      Park Place       |                                   []                                   |
|      Luxury Tax       |                                   []                                   |
|       Boardwalk       |                                   []                                   |
Player 1 ($1500)  
Player 2 ($1500)  
Player 3 ($1500)  
Player 4 ($1500)  
Player 5 ($1500)  
Player 6 ($1500)  
Player 7 ($1500)  
current player: 1

Frame #1:
|          GO           | [Player 2, Player 3, Player 4, Player 5, Player 6, Player 7] |
| Mediterranean Avenue  |                              []                              |
|    Community Chest    |                              []                              |
|     Baltic Avenue     |                              []                              |
|      Income Tax       |                              []                              |
|   Reading Railroad    |                              []                              |
|    Oriental Avenue    |                              []                              |
|        Chance         |                          [Player 1]                          |
|    Vermont Avenue     |                              []                              |
|  Connecticut Avenue   |                              []                              |
|         Jail          |                              []                              |
|   St. Charles Place   |                              []                              |
|   Electric Company    |                              []                              |
|     States Avenue     |                              []                              |
|    Virginia Avenue    |                              []                              |
| Pennsylvania Railroad |                              []                              |
|    St. James Place    |                              []                              |
|    Community Chest    |                              []                              |
|   Tennessee Avenue    |                              []                              |
|    New York Avenue    |                              []                              |
|     Free Parking      |                              []                              |
|    Kentucky Avenue    |                              []                              |
|        Chance         |                              []                              |
|    Indiana Avenue     |                              []                              |
|    Illinois Avenue    |                              []                              |
|   B. & O. Railroad    |                              []                              |
|    Atlantic Avenue    |                              []                              |
|    Ventnor Avenue     |                              []                              |
|      Water Works      |                              []                              |
|    Marvin Gardens     |                              []                              |
|      Go To Jail       |                              []                              |
|    Pacific Avenue     |                              []                              |
| North Carolina Avenue |                              []                              |
|    Community Chest    |                              []                              |
|  Pennsylvania Avenue  |                              []                              |
|  Short Line Railroad  |                              []                              |
|        Chance         |                              []                              |
|      Park Place       |                              []                              |
|      Luxury Tax       |                              []                              |
|       Boardwalk       |                              []                              |
Player 1 ($1550)  
Player 2 ($1500)  
Player 3 ($1500)  
Player 4 ($1500)  
Player 5 ($1500)  
Player 6 ($1500)  
Player 7 ($1500)  
current player: 2

Frame #2:
|          GO           | [Player 3, Player 4, Player 5, Player 6, Player 7] |
| Mediterranean Avenue  |                         []                         |
|    Community Chest    |                         []                         |
|     Baltic Avenue     |                         []                         |
|      Income Tax       |                         []                         |
|   Reading Railroad    |                         []                         |
|    Oriental Avenue    |                         []                         |
|        Chance         |                     [Player 1]                     |
|    Vermont Avenue     |                         []                         |
|  Connecticut Avenue   |                         []                         |
|         Jail          |                         []                         |
|   St. Charles Place   |                     [Player 2]                     |
|   Electric Company    |                         []                         |
|     States Avenue     |                         []                         |
|    Virginia Avenue    |                         []                         |
| Pennsylvania Railroad |                         []                         |
|    St. James Place    |                         []                         |
|    Community Chest    |                         []                         |
|   Tennessee Avenue    |                         []                         |
|    New York Avenue    |                         []                         |
|     Free Parking      |                         []                         |
|    Kentucky Avenue    |                         []                         |
|        Chance         |                         []                         |
|    Indiana Avenue     |                         []                         |
|    Illinois Avenue    |                         []                         |
|   B. & O. Railroad    |                         []                         |
|    Atlantic Avenue    |                         []                         |
|    Ventnor Avenue     |                         []                         |
|      Water Works      |                         []                         |
|    Marvin Gardens     |                         []                         |
|      Go To Jail       |                         []                         |
|    Pacific Avenue     |                         []                         |
| North Carolina Avenue |                         []                         |
|    Community Chest    |                         []                         |
|  Pennsylvania Avenue  |                         []                         |
|  Short Line Railroad  |                         []                         |
|        Chance         |                         []                         |
|      Park Place       |                         []                         |
|      Luxury Tax       |                         []                         |
|       Boardwalk       |                         []                         |
Player 1 ($1550)  
Player 2 ($1360) St. Charles Place 
Player 3 ($1500)  
Player 4 ($1500)  
Player 5 ($1500)  
Player 6 ($1500)  
Player 7 ($1500)  
current player: 3

Frame #3:
|          GO           | [Player 4, Player 5, Player 6, Player 7] |
| Mediterranean Avenue  |                    []                    |
|    Community Chest    |                    []                    |
|     Baltic Avenue     |                    []                    |
|      Income Tax       |                [Player 3]                |
|   Reading Railroad    |                    []                    |
|    Oriental Avenue    |                    []                    |
|        Chance         |                [Player 1]                |
|    Vermont Avenue     |                    []                    |
|  Connecticut Avenue   |                    []                    |
|         Jail          |                    []                    |
|   St. Charles Place   |                [Player 2]                |
|   Electric Company    |                    []                    |
|     States Avenue     |                    []                    |
|    Virginia Avenue    |                    []                    |
| Pennsylvania Railroad |                    []                    |
|    St. James Place    |                    []                    |
|    Community Chest    |                    []                    |
|   Tennessee Avenue    |                    []                    |
|    New York Avenue    |                    []                    |
|     Free Parking      |                    []                    |
|    Kentucky Avenue    |                    []                    |
|        Chance         |                    []                    |
|    Indiana Avenue     |                    []                    |
|    Illinois Avenue    |                    []                    |
|   B. & O. Railroad    |                    []                    |
|    Atlantic Avenue    |                    []                    |
|    Ventnor Avenue     |                    []                    |
|      Water Works      |                    []                    |
|    Marvin Gardens     |                    []                    |
|      Go To Jail       |                    []                    |
|    Pacific Avenue     |                    []                    |
| North Carolina Avenue |                    []                    |
|    Community Chest    |                    []                    |
|  Pennsylvania Avenue  |                    []                    |
|  Short Line Railroad  |                    []                    |
|        Chance         |                    []                    |
|      Park Place       |                    []                    |
|      Luxury Tax       |                    []                    |
|       Boardwalk       |                    []                    |
Player 1 ($1550)  
Player 2 ($1360) St. Charles Place 
Player 3 ($1300)  
Player 4 ($1500)  
Player 5 ($1500)  
Player 6 ($1500)  
Player 7 ($1500)  
current player: 4

Frame #4:
|          GO           | [Player 5, Player 6, Player 7] |
| Mediterranean Avenue  |               []               |
|    Community Chest    |               []               |
|     Baltic Avenue     |               []               |
|      Income Tax       |           [Player 3]           |
|   Reading Railroad    |               []               |
|    Oriental Avenue    |               []               |
|        Chance         |      [Player 1, Player 4]      |
|    Vermont Avenue     |               []               |
|  Connecticut Avenue   |               []               |
|         Jail          |               []               |
|   St. Charles Place   |           [Player 2]           |
|   Electric Company    |               []               |
|     States Avenue     |               []               |
|    Virginia Avenue    |               []               |
| Pennsylvania Railroad |               []               |
|    St. James Place    |               []               |
|    Community Chest    |               []               |
|   Tennessee Avenue    |               []               |
|    New York Avenue    |               []               |
|     Free Parking      |               []               |
|    Kentucky Avenue    |               []               |
|        Chance         |               []               |
|    Indiana Avenue     |               []               |
|    Illinois Avenue    |               []               |
|   B. & O. Railroad    |               []               |
|    Atlantic Avenue    |               []               |
|    Ventnor Avenue     |               []               |
|      Water Works      |               []               |
|    Marvin Gardens     |               []               |
|      Go To Jail       |               []               |
|    Pacific Avenue     |               []               |
| North Carolina Avenue |               []               |
|    Community Chest    |               []               |
|  Pennsylvania Avenue  |               []               |
|  Short Line Railroad  |               []               |
|        Chance         |               []               |
|      Park Place       |               []               |
|      Luxury Tax       |               []               |
|       Boardwalk       |               []               |
Player 1 ($1550)  
Player 2 ($1360) St. Charles Place 
Player 3 ($1300)  
Player 4 ($1485)  
Player 5 ($1500)  
Player 6 ($1500)  
Player 7 ($1500)  
current player: 5

Frame #5:
|          GO           |      [Player 6, Player 7]      |
| Mediterranean Avenue  |               []               |
|    Community Chest    |               []               |
|     Baltic Avenue     |               []               |
|      Income Tax       |           [Player 3]           |
|   Reading Railroad    |               []               |
|    Oriental Avenue    |               []               |
|        Chance         | [Player 1, Player 4, Player 5] |
|    Vermont Avenue     |               []               |
|  Connecticut Avenue   |               []               |
|         Jail          |               []               |
|   St. Charles Place   |           [Player 2]           |
|   Electric Company    |               []               |
|     States Avenue     |               []               |
|    Virginia Avenue    |               []               |
| Pennsylvania Railroad |               []               |
|    St. James Place    |               []               |
|    Community Chest    |               []               |
|   Tennessee Avenue    |               []               |
|    New York Avenue    |               []               |
|     Free Parking      |               []               |
|    Kentucky Avenue    |               []               |
|        Chance         |               []               |
|    Indiana Avenue     |               []               |
|    Illinois Avenue    |               []               |
|   B. & O. Railroad    |               []               |
|    Atlantic Avenue    |               []               |
|    Ventnor Avenue     |               []               |
|      Water Works      |               []               |
|    Marvin Gardens     |               []               |
|      Go To Jail       |               []               |
|    Pacific Avenue     |               []               |
| North Carolina Avenue |               []               |
|    Community Chest    |               []               |
|  Pennsylvania Avenue  |               []               |
|  Short Line Railroad  |               []               |
|        Chance         |               []               |
|      Park Place       |               []               |
|      Luxury Tax       |               []               |
|       Boardwalk       |               []               |
Player 1 ($1600)  
Player 2 ($1410) St. Charles Place 
Player 3 ($1350)  
Player 4 ($1535)  
Player 5 ($1200)  
Player 6 ($1550)  
Player 7 ($1550)  
current player: 6

Frame #6:
|          GO           |                [Player 7]                |
| Mediterranean Avenue  |                    []                    |
|    Community Chest    |                    []                    |
|     Baltic Avenue     |                    []                    |
|      Income Tax       |                [Player 3]                |
|   Reading Railroad    |                    []                    |
|    Oriental Avenue    |                    []                    |
|        Chance         | [Player 1, Player 4, Player 5, Player 6] |
|    Vermont Avenue     |                    []                    |
|  Connecticut Avenue   |                    []                    |
|         Jail          |                    []                    |
|   St. Charles Place   |                [Player 2]                |
|   Electric Company    |                    []                    |
|     States Avenue     |                    []                    |
|    Virginia Avenue    |                    []                    |
| Pennsylvania Railroad |                    []                    |
|    St. James Place    |                    []                    |
|    Community Chest    |                    []                    |
|   Tennessee Avenue    |                    []                    |
|    New York Avenue    |                    []                    |
|     Free Parking      |                    []                    |
|    Kentucky Avenue    |                    []                    |
|        Chance         |                    []                    |
|    Indiana Avenue     |                    []                    |
|    Illinois Avenue    |                    []                    |
|   B. & O. Railroad    |                    []                    |
|    Atlantic Avenue    |                    []                    |
|    Ventnor Avenue     |                    []                    |
|      Water Works      |                    []                    |
|    Marvin Gardens     |                    []                    |
|      Go To Jail       |                    []                    |
|    Pacific Avenue     |                    []                    |
| North Carolina Avenue |                    []                    |
|    Community Chest    |                    []                    |
|  Pennsylvania Avenue  |                    []                    |
|  Short Line Railroad  |                    []                    |
|        Chance         |                    []                    |
|      Park Place       |                    []                    |
|      Luxury Tax       |                    []                    |
|       Boardwalk       |                    []                    |
Player 1 ($1600)  
Player 2 ($1410) St. Charles Place 
Player 3 ($1350)  
Player 4 ($1535)  
Player 5 ($1200)  
Player 6 ($1550)  GetOutOfJailFree
Player 7 ($1550)  
current player: 7

Frame #7:
|          GO           |                    []                    |
| Mediterranean Avenue  |                    []                    |
|    Community Chest    |                    []                    |
|     Baltic Avenue     |                    []                    |
|      Income Tax       |                [Player 3]                |
|   Reading Railroad    |                    []                    |
|    Oriental Avenue    |                    []                    |
|        Chance         | [Player 1, Player 4, Player 5, Player 6] |
|    Vermont Avenue     |                    []                    |
|  Connecticut Avenue   |                    []                    |
|         Jail          |                    []                    |
|   St. Charles Place   |                [Player 2]                |
|   Electric Company    |                    []                    |
|     States Avenue     |                    []                    |
|    Virginia Avenue    |                    []                    |
| Pennsylvania Railroad |                [Player 7]                |
|    St. James Place    |                    []                    |
|    Community Chest    |                    []                    |
|   Tennessee Avenue    |                    []                    |
|    New York Avenue    |                    []                    |
|     Free Parking      |                    []                    |
|    Kentucky Avenue    |                    []                    |
|        Chance         |                    []                    |
|    Indiana Avenue     |                    []                    |
|    Illinois Avenue    |                    []                    |
|   B. & O. Railroad    |                    []                    |
|    Atlantic Avenue    |                    []                    |
|    Ventnor Avenue     |                    []                    |
|      Water Works      |                    []                    |
|    Marvin Gardens     |                    []                    |
|      Go To Jail       |                    []                    |
|    Pacific Avenue     |                    []                    |
| North Carolina Avenue |                    []                    |
|    Community Chest    |                    []                    |
|  Pennsylvania Avenue  |                    []                    |
|  Short Line Railroad  |                    []                    |
|        Chance         |                    []                    |
|      Park Place       |                    []                    |
|      Luxury Tax       |                    []                    |
|       Boardwalk       |                    []                    |
Player 1 ($1600)  
Player 2 ($1410) St. Charles Place 
Player 3 ($1350)  
Player 4 ($1535)  
Player 5 ($1200)  
Player 6 ($1550)  GetOutOfJailFree
Player 7 ($1550)  
current player: 1